src/shared/common_lib.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+# Common functions that are shared across the different libraries
+
+CICD_TOOLS_COMMON_LOADED=${CICD_TOOLS_COMMON_LOADED:-1}
+LOCAL_BUILD=${LOCAL_BUILD:-false}
+
+if [ "$CICD_TOOLS_COMMON_LOADED" -eq 0 ]; then
+  cicd_tools::debug "common library already loaded, skipping"
+  return 0
+fi
+
+if [ -z "$CICD_TOOLS_SCRIPTS_DIR" ]; then
+  echo "scripts directory not defined, please load through main.sh script" >&2
```

Comment:
Re this error, if I hit this error I would not know exactly what to do... the docs only talk about using `bootstrap.sh`, right? Should a user ever be calling `main.sh` themselves?

Reply:
Mm.. This I acknowledge is in part due to a poor design choice, where the "SCRIPTS DIR" is set or not is the condition to "detect" if the file was sourced directly, or through `main.sh`

Initially it was only the `bootstrap.sh` script, that did all of it, clone the repo and prepare to load the modules, then I split it in two parts, the `bootstrap` only required for end users to consume the library externally, and `main` being the actual entrypoint - so "all bootstrap does" is sourcing 'main'  , which does some setup, including checking the "scripts dir" is present and available. 

If you are developing the library, and have all the scripts available to source, you could technically do it, but it would mess up the setup phase, so the entrypoint has to be always `main.sh`  - that's the context for that message.

Bootstrap script is only required if you have nothing, but if you have the library, then `main.sh` is all you need. 

Probably this requires improving the way the documentation describes how to use the library, and what are `bootstrap` and `main` scripts meant to be used for.

To include a contributing section is indeed something I want to do but not until I extend this to all scripts within this repository (I need to port and / or deprecate parts of the old scripts too)

To answer to your question - yes, the users would be always sourcing main, however indirectly (through bootstrap) if they're not maintaining the library.

I think I'd change the way the library detects how it was loaded to a variable in `main` and add an optional "or through `bootstrap.sh` " bit to clarify. 

please LMK if you have additional thoughts following me providing this context.

Reply:
OK, right, I do think tweaking the docs to talk about the purpose of 'main.sh' vs 'bootstrap.sh' will make this clearer

Reply:
I've added [a new section in the README and reworked the previous block](https://github.com/RedHatInsights/cicd-tools/pull/22/commits/59fe7b07c15a80f3668dde4e43e4019428693fa1) to clarify this, please take a look, don't hesitate to suggest anything if it reads funny or anything :)